lua/neo-tree/sources/common/components.lua

```diff
@@ -74,6 +74,45 @@ M.current_filter = function(config, node, state)
   }
 end
 
+---`sign_getdefined` based wrapper with compatibility
+---@param severity string
+---@return vim.fn.sign_getdefined.ret.item
+local function get_defined_sign(severity)
+  local defined
+
+  if vim.fn.has("nvim-0.10") then
+    local signs_config = vim.diagnostic.config().signs
+    if type(signs_config) == "table" then
+      local identifier = severity:sub(1, 1)
+      if identifier == "H" then
+        identifier = "N"
+      end
+      defined = {
+        text = signs_config.text[vim.diagnostic.severity[identifier]],
+        texthl = "DiagnosticSign" .. severity,
+      }
+    end
+  else -- before 0.10
+    defined = vim.fn.sign_getdefined("DiagnosticSign" .. severity)
+    if not defined then
```

Comment:
Same; In lua, an empty table `{}` evaluates to true, so this will be never executed.

Reply:
This branch is extracted from the original, it's maybe an undiscovered bug.